validator/block_validator.go

```diff
@@ -222,23 +302,58 @@ func (v *BlockValidator) readLastBlockValidatedDbInfo(reorgingToBlock *types.Blo
 	return nil
 }
 
-func (v *BlockValidator) prepareBlock(ctx context.Context, header *types.Header, prevHeader *types.Header, msg arbstate.MessageWithMetadata, validationStatus *validationStatus) {
-	preimages, readBatchInfo, hasDelayedMessage, delayedMsgToRead, err := BlockDataForValidation(ctx, v.blockchain, v.inboxReader, header, prevHeader, msg, v.config().StorePreimages)
-	if err != nil {
-		log.Error("failed to set up validation", "err", err, "header", header, "prevHeader", prevHeader)
-		return
+func (v *BlockValidator) sendRecord(s *validationStatus, mustDeref bool) error {
+	if !v.Started() {
+		// this could only be sent by NewBlock, so mustDeref is not sent
+		return nil
+	}
+	prevHeader := s.Entry.PrevBlockHeader
+	if prevHeader == nil {
+		mustDeref = false
+	}
+	if !s.replaceStatus(Unprepared, RecordSent) {
+		if mustDeref {
+			v.recordingDatabase.Dereference(prevHeader)
+		}
+		return errors.Errorf("failed status check for send record. Status: %v", s.getStatus())
 	}
-	validationEntry, err := newValidationEntry(prevHeader, header, hasDelayedMessage, delayedMsgToRead, preimages, readBatchInfo)
+	v.LaunchThread(func(ctx context.Context) {
+		if mustDeref {
+			defer v.recordingDatabase.Dereference(prevHeader)
+		}
+		err := v.ValidationEntryRecord(ctx, s.Entry, true)
+		if ctx.Err() != nil {
+			return
+		}
+		if err != nil {
+			log.Error("Error while recording", "err", err)
+			return
```

Comment:
Is this error recoverable? I think we should try to make it recoverable, but if not it should probably be fatal.